cypress/integration/search_spec.js

```diff
@@ -1,25 +1,109 @@
+            cy.search(good_term);
+            cy.url().should('include', `/results?searchTerm=${good_term.replace(/\s+/g, '+')}`);
+        });
+
+        it('display message when no results are available', function () {
+            cy.search(bad_term);
+            cy.contains(`No results found for "${bad_term}`);
+        });
+
+        it('display result counts', function () {
+            cy.search(good_term);
+            cy.get('#resultsCount');
+        });
 
-            // TODO: add tests around results returned
+        it('have a show more button when there are >20 results', function () {
+            cy.search(good_term);
+            cy.get('#moreButton').click();
         });
 
-        it('handles queries with no results', function () {
-            // TODO: implement
+        it('do not have show more button for no results', function () {
+            cy.search(bad_term);
+            cy.get('#moreButton').should('not.exist');
         });
 
-    });
+        it('include a link for every row', function () {
+            cy.search(good_term);
+            cy.get(".article-title").click({ multiple: true, force: true})
+            cy.get('.pmids')
+            .each(function (el) {
+                expect(el.context).to.have.property('href')
+            })
+        })
+
+        it('include authors for every row', function () {
+            cy.search(good_term);
+            cy.get(".article-title").click({ multiple: true, force: true})
+            cy.get('.pm-list-row-authors')
+            .each(function (el) {
+                console.log(el.context)
+            })
+        })
+
+        it('include a close button for every row', function () {
+            cy.search(good_term);
+            cy.get(".article-title").click({ multiple: true, force: true })
+            cy.get('.close')
+            .each(function (el) {
+                console.log(el.context)
+            })
+        })
+
+        it('include checkbox for every item', function () {
+            cy.search(good_term);
+            cy.get(".article-title").click({ multiple: true, force: true })
+            cy.get('.uid')
+            .each(function (el) {
+                expect(el.context.type).equals('checkbox')
+            })
+        })
+
+        it('have additional information attached', function () {
+            cy.search(good_term);
+            cy.get(".article-title").click({ multiple: true, force: true })
+            cy.get('.list-group-item-header').find('.list-view-pf-additional-info-item')
+        })
+
+        it('has a menu button and export button', function(){
+            cy.search(good_term);
+            cy.get("#menuButton").click();
+            cy.get(".applauncher-pf-link").click({ multiple: true, force: true });
+        })
+
+        it('results have some information item element [abstract,sentence,table]', function () {
+            cy.search(good_term);
+            cy.get(".article-title").click({ multiple: true, force: true })
```

Comment:
for functions that you invoke multiple times, like "clicking article title" you should consider adding a custom `cypress` function as was done for `search(term)` for example